app/src/org/commcare/android/adapters/EntityListAdapter.java

```diff
@@ -254,7 +254,7 @@ private void search() {
                         String field = e.getNormalizedField(i);
                         if (field != "" && field.toLowerCase(currentLocale).contains(filter)) {
                             add = true;
-                            continue filter;
```

Comment:
IF continuing, the *add* variable would be set to false, if the next search term doesn't match, it will end up here https://github.com/dimagi/commcare-odk/pull/867/files#diff-2a0c2809914e40cd56d9b605f3d6c0c5L273